### PR TITLE
US14589 - Persistent Logging

### PR DESCRIPTION
**Pull request by jpc11623**
Commit of WIP logging approach

### CODE REVIEW

File: Crossroads.Service.Finance/Services/Pushpay/PushpayService.cs [hunk 1]
```diff
@@ -267,6 +268,19 @@ private MpRecurringGift BuildAndCreateNewRecurringGift (PushpayRecurringGiftDto
             // cancel the recurring gift in stripe, if exists
             if (pushpayRecurringGift.Notes != null && pushpayRecurringGift.Notes.Trim().StartsWith("sub_", StringComparison.Ordinal))
             {
+                var eventData = new Dictionary<string, object>();
+                eventData.Add("Donor First Name", pushpayRecurringGift.Payer.FirstName);
```
Comment (dillon-courts) [inline Crossroads.Service.Finance/Services/Pushpay/PushpayService.cs]: I think New Relic requires the keys to be camelCase.

File: Crossroads.Service.Finance/Services/Pushpay/PushpayService.cs [hunk 2]
```diff
@@ -267,6 +268,19 @@ private MpRecurringGift BuildAndCreateNewRecurringGift (PushpayRecurringGiftDto
             // cancel the recurring gift in stripe, if exists
             if (pushpayRecurringGift.Notes != null && pushpayRecurringGift.Notes.Trim().StartsWith("sub_", StringComparison.Ordinal))
             {
+                var eventData = new Dictionary<string, object>();
+                eventData.Add("Donor First Name", pushpayRecurringGift.Payer.FirstName);
+                eventData.Add("Donor Last Name", pushpayRecurringGift.Payer.FirstName);
```
Comment (dillon-courts) [inline Crossroads.Service.Finance/Services/Pushpay/PushpayService.cs]: I imagine this should be Payer.LastName

**Comment by timmyg** [comment]
I think something like this would be *real nice* (cousin eddie voice from christmas vacation)
```
var log = Logger.init(LogEventType.StripeCancel)

log.Push("Donor First Name", pushpayRecurringGift.Payer.FirstName)
log.Push("Donor Email Address", pushpayRecurringGift.Payer.EmailAddress)

log.Persist()
```

File: Crossroads.Service.Finance/Services/Pushpay/PushpayService.cs [hunk 3]
```diff
@@ -267,6 +270,13 @@ private MpRecurringGift BuildAndCreateNewRecurringGift (PushpayRecurringGiftDto
             // cancel the recurring gift in stripe, if exists
             if (pushpayRecurringGift.Notes != null && pushpayRecurringGift.Notes.Trim().StartsWith("sub_", StringComparison.Ordinal))
             {
+                var logEventEntry = new LogEventEntry(LogEventType.StripeCancel);
+                logEventEntry.Push("Donor First Name", pushpayRecurringGift.Payer.FirstName);
+                logEventEntry.Push("Donor First Name", pushpayRecurringGift.Payer.LastName);
```
Comment (briancavanaugh) [inline Crossroads.Service.Finance/Services/Pushpay/PushpayService.cs]: Should the first param be "Last Name"?

File: Utilities/Logging/LoggingEnums.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Utilities.Logging
+{
+    public enum LogEventType
+    {
+        stripeCancel, noDepositsToSync, depositsCreatedCount, incomingPushpayWebhook, donationNotFoundForTransaction,
```
Comment (briancavanaugh) [inline Utilities/Logging/LoggingEnums.cs]: To make this more readable, to find entries quicker, and to add new ones later, I would suggest alphabetizing and have each entry on it's own line.

**Pull request merged**